test-common/test-common-runtime/src/main/java/org/apache/kafka/common/test/KafkaClusterTestKit.java

```diff
@@ -477,53 +487,43 @@ private void formatNode(
             }
-            if (nodes.bootstrapMetadata().featureLevel(KRaftVersion.FEATURE_NAME) > 0) {
-                StringBuilder dynamicVotersBuilder = new StringBuilder();
-                String prefix = "";
-                if (standalone) {
-                    if (nodeId == TestKitDefaults.CONTROLLER_ID_OFFSET) {
-                        final var controllerNode = nodes.controllerNodes().get(nodeId);
-                        dynamicVotersBuilder.append(
-                            String.format(
-                                "%d@localhost:%d:%s",
-                                controllerNode.id(),
-                                socketFactoryManager.
-                                    getOrCreatePortForListener(controllerNode.id(), controllerListenerName),
-                                controllerNode.metadataDirectoryId()
-                            )
-                        );
-                        formatter.setInitialControllers(DynamicVoters.parse(dynamicVotersBuilder.toString()));
-                    } else {
-                        formatter.setNoInitialControllersFlag(true);
-                    }
-                } else if (initialVoterSet.isPresent()) {
-                    for (final var controllerNode : initialVoterSet.get().entrySet()) {
-                        final var voterId = controllerNode.getKey();
-                        final var voterDirectoryId = controllerNode.getValue();
-                        dynamicVotersBuilder.append(prefix);
-                        prefix = ",";
-                        dynamicVotersBuilder.append(
-                            String.format(
-                                "%d@localhost:%d:%s",
-                                voterId,
-                                socketFactoryManager.
-                                    getOrCreatePortForListener(voterId, controllerListenerName),
-                                voterDirectoryId
-                            )
-                        );
-                    }
-                    formatter.setInitialControllers(DynamicVoters.parse(dynamicVotersBuilder.toString()));
-                } else {
-                    for (TestKitNode controllerNode : nodes.controllerNodes().values()) {
-                        int port = socketFactoryManager.
-                            getOrCreatePortForListener(controllerNode.id(), controllerListenerName);
-                        dynamicVotersBuilder.append(prefix);
-                        prefix = ",";
-                        dynamicVotersBuilder.append(String.format("%d@localhost:%d:%s",
-                            controllerNode.id(), port, controllerNode.metadataDirectoryId()));
-                    }
+            StringBuilder dynamicVotersBuilder = new StringBuilder();
+            String prefix = "";
+            if (standalone) {
+                if (nodeId == TestKitDefaults.CONTROLLER_ID_OFFSET) {
```

Comment:
The true port of the standalone controller is `TestKitDefaults.BROKER_ID_OFFSET + TestKitDefaults.CONTROLLER_ID_OFFSET`. We will likely not change the value of `BROKER_ID_OFFSET` in the future, but should we still use `BROKER_ID_OFFSET + CONTROLLER_ID_OFFSET` instead?

https://github.com/apache/kafka/blob/trunk/test-common/test-common-runtime/src/main/java/org/apache/kafka/common/test/TestKitNodes.java#L175